### ISSUE TITLE
Frustrated with the Program

### ISSUE THREAD

**Issue by bilalinamdar** [BoostIO/Boostnote#3720]
Dear Team,

Why the program on every update changes drastically ?
I mean i was the early adopter.

First it was Cloud storage.
Than upgrade asked for Local storage 
Now from local storage (depreciated) to boostcloud local next.

Why make enduser suffer which drastic big change in the main operating of the software ?

I really look great potential in this project but again i am too frustrated to make it my regular app as any new changes might erase all my data and always have to adapt to new env.

**Comment by Komediruzecki** [comment]
Hi, bilalinamdar,

I get your frustration, from now on it should not be any drastical changes as cloud and local apps are separated.

If I understood correctly, you first had cloud (legacy cloud and/or local storage)
Then we discarded that because new boosthub/new cloud arrived (free/pro plans)
Then to use the free local storage you need to transition from legacy cloud as it was not synced with servers anymore
After that the local storage was separated into separate desktop application 
So if you had transitioned to local storage and not new cloud you would need to install new desktop app for local storage (and import storage location there)

The new cloud now offers unlimited documents on free plan and other pro features for paid plans
The new desktop boost note local only app provides support only for local storage (no cloud, no sync, no analytics etc.)

I agree with you it was a rough year for the user, but this was necessary to adjust to more flexible environment and separate the concerns, as with anything great, if you want it fast, you need to change and start over many times.

But on the other hand, I don't agree we changed it with every update and in short timespan, the above transitions did happen from start of the year to now so it was almost 8 months of updates and transitions (alongside other improvements and fixes).

I definitely hear you but can assure you there should not be any such drastic change made further on, the only thing that did change frequently was handling of cloud storages, since legacy cloud was deprecated due to new cloud release, we needed to drop those storages and support local file system storage only and new cloud.

If you have any specific questions, further feedback or interested in further development plans, please don't hesitate to write!

**Comment by bilalinamdar** [comment]
I understand you concern too.
I am a fan of this kind of software and my first prefrence was snippets (discontinued), than masscode, than boostnote.
I usually work on onenote.
What I mean to say is whenever a user open the app he need to put or get something. Meanwhile the update give headache of transforming the entire sotrage scene.
Never mind I truly hope major change will have a notification or some sort of changes detail on the UPDATE? diaglog. Atleast we will be aware that after clicking update we are not going in rabbit hole.
BTW I will be using this software and i congratulate you on such a huge jump since 2020.

**Comment by Komediruzecki** [comment]
I agree on the changelog, currently the implementation only updates the app on Windows (not sure how ti works on macOS), but on linux the updater does not work, so you need to manually go to the github release and there you can see a changelog: see here for example https://github.com/BoostIO/BoostNote.next/releases/tag/v0.21.1 (yes it should also be improved)

I think that adding a changelog greatly improves the software updates procedure, so I will probably add it myself for the next local space application release, but for that to work, I need to fix the Linux updater as well.

I was using the desktop app for quite some time, if you have latest release (local space only or cloud desktop app) it should not be slow (cloud space editor was a bit slow few releases back), can you please describe if this is still an issue and closely describe which part of the application is slow?

Thank you once again for the feedback, please let me know if you have any more questions or concerns.

**Comment by armartin** [comment]
Agree, this change has made it impossible to do any work and instead I've had to slog through manually migrating notes today given an abysmal forced migration process. Of 1469 notes (most of which were clunky imports from Evernote that I abandoned realizing I'd just need to go back to Evernote to see again), 532 failed to import of which 63 were fully written **using the Boostnote's app!** Before forcing the storage migration, it'd be super helpful to debug migration issues first. So frustrating!

![image](https://user-images.githubusercontent.com/2180892/129085715-6fbc7463-17a6-4475-a7f3-d280398b4947.png)

**Comment by Komediruzecki** [comment]
> Agree, this change has made it impossible to do any work and instead I've had to slog through manually migrating notes today given an abysmal forced migration process. Of 1469 notes (most of which were clunky imports from Evernote that I abandoned realizing I'd just need to go back to Evernote to see again), 532 failed to import of which 63 were fully written **using the Boostnote's app!** Before forcing the storage migration, it'd be super helpful to debug migration issues first. So frustrating!
> 
> ![image](https://user-images.githubusercontent.com/2180892/129085715-6fbc7463-17a6-4475-a7f3-d280398b4947.png)

Hi, armartin,

I have done migrations many times and most of the time, the migration actually successfully transfered the notes but reported that it didn't, you should have a list of those notes that failed on this or next page, so be sure to look it up and see if they really failed to migrate.

Other than that, I can suggest first transfering all the notes to file system storage and then migrating to cloud. 
Did you migrate notes from local file system storage or local storage (legacy cloud) storage?

**Comment by armartin** [comment]
Hi Komediruzecki,

Thanks for checking in! I checked the notes that were described as unsuccessful and they unfortunately were indeed missing and failed to migrate. I migrated from a local file system storage (not legacy cloud).

I fully appreciate not having to migrate to cloud - it's a helpful option to keep around local storage (though it would be easier if downloading a 2nd app to see local notes wasn't required). Since these are work notes though, it was important to me at least to ensure backup capability, so the cloud storage is indeed a very much appreciated upgrade! (Was just hoping for an easier transition). Thanks for your suggestions.

**Comment by Komediruzecki** [comment]
> 
> 
> Hi Komediruzecki,
> 
> Thanks for checking in! I checked the notes that were described as unsuccessful and they were unfortunately indeed missing and failed to migrate. I migrated from a local file system storage (not legacy cloud).
> 
> I fully appreciate not having to migrate to cloud - it's a helpful option to keep around local storage (though it would be easier if downloading a 2nd app to see local notes wasn't required). Since these are work notes though, it was important to me at least to ensure backup capability, so the cloud storage is indeed a very much appreciated upgrade! (Was just hoping for an easier transition). Thanks for your suggestions.

No problem,

yes, the cloud is great, and if you want to keep using local storage, we separated it to another application so that it is easier to maintain.

For migration to cloud, when I migrated my notes, the only notes which it showed as not migrated were the ones which had some kind of problems/issues inside, some weren't synced correctly in legacy cloud, some were not having proper attachments etc. But all of those notes actually did migrate correctly.

If you have any other questions or concerns, please don't hesitate to ask!
